chainer_chemistry/utils/extend.py

```diff
@@ -0,0 +1,66 @@
+from collections import Iterable
+
+import numpy
+import six
+from chainer import cuda
+
+
+def _to_list(a):
+    if isinstance(a, Iterable):
+        a = list(a)
+    else:
+        a = [a]
+    return a
+
+
+def extend_node(node, out_size, axis=-1, value=0):
+    return extend_arrays_to_size(
+        node, out_size=out_size, axis=axis, value=value)
+
+
+def extend_adj(adj, out_size, axis=None, value=0):
+    axis = axis or [-1, -2]
+    return extend_arrays_to_size(
+        adj, out_size=out_size, axis=axis, value=value)
+
+
+def extend_arrays_to_size(arrays, out_size, axis=-1, value=0):
+    """Extend size of `node` array
+
+    Args:
+        arrays (numpy.ndarray): the array whose `axis` to be extended.
+            first axis is considered as "batch" axis.
+        out_size (int): target output size for specified `axis`.
+        axis (int): node feature axis to be extended.
+        value (int or float): value to be filled for extended place.
+
+    Returns (numpy.ndarray): extended `node` array, extended place is filled
+        with `value`
+
+    """
+    batch_size = len(arrays)
+
+    in_shape = _to_list(arrays[0].shape)
+    out_shape = [batch_size] + in_shape
+
+    axis = _to_list(axis)
```

Comment:
I feel if axis contains `0`, raising warning is kind  to user. But it is not necessary to solve.

Reply:
ok